src/unix/taskbarx11.cpp

```diff
@@ -131,25 +132,26 @@ wxTaskBarIconArea::wxTaskBarIconArea(wxTaskBarIcon *icon, const wxBitmap &bmp)
     }
 }
 
-void wxTaskBarIconArea::SetTrayIcon(const wxBitmap& bmp)
+void wxTaskBarIconArea::SetTrayIcon(const wxBitmapBundle& bmp)
 {
     m_bmp = bmp;
+    m_bmpReal = m_bmp.GetBitmapFor(this);
 
     // determine suitable bitmap size:
     wxSize winsize(GetClientSize());
-    wxSize bmpsize(m_bmp.GetWidth(), m_bmp.GetHeight());
+    wxSize bmpsize(m_bmpReal.GetLogicalWidth(), m_bmpReal.GetLogicalHeight());
     wxSize iconsize(wxMin(winsize.x, bmpsize.x), wxMin(winsize.y, bmpsize.y));
 
     // rescale the bitmap to fit into the tray icon window:
     if (bmpsize != iconsize)
     {
-        wxImage img = m_bmp.ConvertToImage();
+        wxImage img = m_bmpReal.ConvertToImage();
         img.Rescale(iconsize.x, iconsize.y);
-        m_bmp = wxBitmap(img);
+        m_bmpReal = wxBitmap(img);
```

Comment:
I think we should change this code (maybe in a separate commit just in case there are some problems with doing it like this that I don't see) to get bitmap of `iconsize` from the bundle directly.